vertical-pod-autoscaler/pkg/recommender/model/vpa_test.go

```diff
@@ -273,66 +276,136 @@ func TestUseAggregationIfMatching(t *testing.T) {
 			updateMode:                  &modeAuto,
 			container:                   "second-container",
 			containerLabels:             map[string]string{"different": "labels"},
-			isUnderVPA:                  false,
+			expectedMatching:            false,
 			expectedAggregations:        []string{"test-container"},
 			expectedNeedsRecommendation: false,
 			expectedUpdateMode:          nil,
+		}, {
+			name:         "New matching aggregation with scaling mode Off",
+			aggregations: []string{"test-container"},
+			vpaSelector:  testSelectorStr,
+			resourcePolicy: &vpa_types.PodResourcePolicy{
+				ContainerPolicies: []vpa_types.ContainerResourcePolicy{
+					{
+						ContainerName: "second-container",
+						Mode:          &scalingModeOff,
+					},
+				},
+			},
+			updateMode:                  &modeAuto,
+			container:                   "second-container",
+			containerLabels:             testLabels,
+			expectedMatching:            true,
+			expectedAggregations:        []string{"test-container", "second-container"},
+			expectedNeedsRecommendation: false,
+			expectedUpdateMode:          &modeAuto,
+		}, {
+			name:         "New matching aggregation with default scaling mode Off",
+			aggregations: []string{"test-container"},
+			vpaSelector:  testSelectorStr,
+			resourcePolicy: &vpa_types.PodResourcePolicy{
+				ContainerPolicies: []vpa_types.ContainerResourcePolicy{
+					{
+						ContainerName: "*",
+						Mode:          &scalingModeOff,
+					},
+				},
+			},
+			updateMode:                  &modeAuto,
+			container:                   "second-container",
+			containerLabels:             testLabels,
+			expectedMatching:            true,
+			expectedAggregations:        []string{"test-container", "second-container"},
+			expectedNeedsRecommendation: false,
+			expectedUpdateMode:          &modeAuto,
+		}, {
+			name:         "New matching aggregation with scaling mode Off with default Auto",
+			aggregations: []string{"test-container"},
+			vpaSelector:  testSelectorStr,
+			resourcePolicy: &vpa_types.PodResourcePolicy{
+				ContainerPolicies: []vpa_types.ContainerResourcePolicy{
+					{
+						ContainerName: "*",
+						Mode:          &scalingModeAuto,
+					},
+					{
+						ContainerName: "second-container",
+						Mode:          &scalingModeOff,
+					},
+				},
+			},
+			updateMode:                  &modeAuto,
+			container:                   "second-container",
+			containerLabels:             testLabels,
+			expectedMatching:            true,
+			expectedAggregations:        []string{"test-container", "second-container"},
+			expectedNeedsRecommendation: false,
```

Comment:
Can we also (easily) check here that the "test-container" still has needsRecommendation == true?

Reply:
Done